client/js/s3/request-signer.js

```diff
@@ -153,10 +154,51 @@ qq.s3.RequestSigner = function(o) {
 
             getEncodedHashedPayload: function(body) {
                 var promise = new qq.Promise(),
-                    reader;
+                    reader,
+                    self = this;
 
                 if (qq.isBlob(body)) {
-                    // TODO hash blob in webworker if this becomes a notable perf issue
+                    // Only if a worker is requested will we load it.
+                    // We will fallback to the inline reader if the worker
+                    // fails to load.
+                    if (options.signatureSpec.useWorker) {
+                        if (!this._worker) {
+                            try {
+                                // allow the worker path to be overwritten.. Eg. we aren't using an inline worker.
+                                var workerUrl = options.signatureSpec.useWorker;
+                                if (workerUrl === true) {
+                                    if (!qq.s3.worker) {
+                                        throw new Error("Missing inline s3 worker.");
+                                    }
+                                    workerUrl = qq.s3.worker();
```

Comment:
Not really fond of the name of this function. It's not obvious that a URL is to be returned here, or what the worker does.